torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -187,6 +187,8 @@ def __init__(
         configs,
         save_cache_hook,
         mutated_arg_names: List[str],  # see [Note: clone mutated buffers]
+        is_inference,
```

Comment:
In the future we may want to have a more refined check where we will still copy to cuda if we know that it is not a high water mark increasing clone. Can we do that from the start ? I think it will also be more clear inline what this is doing.